src/edfi_amt_data_lake/parquet/amt/base/academic_time_period_dim/main.py

```diff
@@ -0,0 +1,176 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import pandas as pd
+from decouple import config
+
+from edfi_amt_data_lake.helper.data_frame_generation_result import (
+    data_frame_generation_result,
+)
+from edfi_amt_data_lake.parquet.Common.functions import getEndpointJson
+from edfi_amt_data_lake.parquet.Common.pandasWrapper import (
+    pdMerge,
+    jsonNormalize,
+    renameColumns,
+    create_parquet_file,
+    addColumnIfNotExists,
+    get_descriptor_code_value_from_uri
+)
+
+ENDPOINT_SESSIONS = "sessions"
+ENDPOINT_GRADING_PERIOD_DESCRIPTORS = "gradingPeriodDescriptors"
+ENDPOINT_TERM_DESCRIPTORS = "termDescriptors"
+ENDPOINT_SCHOOL_YEAR_TYPES = "schoolYearTypes"
+
+RESULT_COLUMNS = [
+    "AcademicTimePeriodKey",
+    "SchoolYear",
+    "SchoolYearName",
+    "IsCurrentSchoolYear",
+    "SchoolKey",
+    "SessionKey",
+    "SessionName",
+    "TermName",
+    "GradingPeriodKey",
+    "GradingPeriodName",
+    "LastModifiedDate"
+]
+
+
+@create_parquet_file
+def academic_time_period_dim_frame(
+    file_name: str,
+    columns: list[str],
+    school_year: int
+) -> pd.DataFrame:
+    file_name = file_name
+    sessions_content = getEndpointJson(ENDPOINT_SESSIONS, config("SILVER_DATA_LOCATION"), school_year)
+    grading_periods_descriptors_content = getEndpointJson(ENDPOINT_GRADING_PERIOD_DESCRIPTORS, config("SILVER_DATA_LOCATION"), school_year)
+    term_descriptors_content = getEndpointJson(ENDPOINT_TERM_DESCRIPTORS, config("SILVER_DATA_LOCATION"), school_year)
+    school_year_types_content = getEndpointJson(ENDPOINT_SCHOOL_YEAR_TYPES, config("SILVER_DATA_LOCATION"), school_year)
+
+    session_normalized = jsonNormalize(
```

Comment:
I think that, if we don't have sessions, we should return an empty parquet file. If that is the case, can you add a validations so when the `dataframe` is empty, we return `None`